lightpath/controller.py

```diff
@@ -527,8 +555,59 @@ def get_device(self, device: NodeName) -> Device:
         -------
         Device
             requested device
+
+        Raises
+        ------
+        DeviceLoadingError
+            if the requested device is in the facility but cannot be loaded
         """
         try:
-            return self.graph.nodes[device]['dev']
+            dev_data = self.graph.nodes[device_name]
         except KeyError:
-            logger.error(f'requested device ({device}) not found')
+            logger.error(f'requested device ({device_name}) not in facility')
+            return
+
+        if dev_data['dev'] is not None:
+            return dev_data['dev']
+        elif dev_data['res'] is not None:
+            # not instantiated yet, create and fill
+            try:
+                dev = dev_data['res'].get()
+                self.graph.nodes[device_name]['dev'] = dev
+                return dev
+            except Exception:
+                logger.error(f'Device {device_name} failed to load, '
+                             'attempting to make a mock device')
+                dev = make_mock_device(dev_data['res'])
```

Comment:
Does this mean that unloadable devices will have a stand-in node? If so, what do these stand-in nodes look like to the user? Are they blocking, for example?

Reply:
They will.  Currently they're set to be removed, since I figured if we have a stand-in node it shouldn't be disruptive.  If we wanted to assume the beam is blocked we could just go back to throwing an error if we can't create the device.  (both would force users to go and fix the device class/happi information/whatever issue)

Though it's probably worth making these nodes very obvious to the user so they know something is wrong.  I'll throw a screenshot up, but currently I just rename the device and have it operate as a simulated valve/crystal.

I'm open to suggestions on this though, the more I think about it the more I move toward the fence.

Reply:
I think your philosophy of being minimally disruptive makes sense. It's possible the main things we can do are startup warnings and assigning some unique color to it... but my initial thought about faking a blocked state I think was misguided. The other thing we might be able to do is pick a silly icon for them so they stand out in the GUI.

Reply:
I need to go through and make the coloring consistent overall as well (inserted mirrors with 1.0 transmission shouldn't be red, etc), so I'll touch up the stand-in gui elements there.

Reply:
I like the idea of an icon that would stand out somehow